### PR TITLE
beam-packages: move wxSupport arg up to package set, add beam_nox 

### PR DESCRIPTION
**Pull request by yorickvP**
###### Motivation for this change

It turns out that rabbitmq does not build with `environment.noXlibs` set, and neither does epmd. It's a bit tedious to get out the packages built without X support, or to even override it manually. After various iterations, I think an argument to beam-packages.nix is the best solution. This allows us to set `beam = beam_nox` to disable X support on all erlang packages.

cc @peterhoeg 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yorickvP** [comment]
It no longer applies, will rebase today. cc @jonringer for review

**Review by happysalada — APPROVED** [review]
LGTM

Just as an aside. I checked a little bit around rabbitmq-server. I couldn't find the commit where erlang was set to erlang_nox. It wasn't there at the beginning. I tested on my machine, and rabbitmq-server builds fine with normal erlang. I checked the docs and it doesn't say anything about a specific version of erlang. I really can't figure out why rabbitmq-server wants to _nox version. If anyone knows, I would be interested.

**Comment by peterhoeg** [comment]
To avoid pulling in wxwidgets dependencies when it isn't necessary. Ideally we would of course be able to exclude *all* X stuff...

File: pkgs/development/interpreters/erlang/generic-builder.nix [hunk 1]
```diff
@@ -3,11 +3,11 @@
 # TODO: use jdk https://github.com/NixOS/nixpkgs/pull/89731
 , openjdk8 ? null # javacSupport
 , unixODBC ? null # odbcSupport
-, libGL ? null, libGLU ? null, wxGTK ? null, wxmac ? null, xorg ? null # wxSupport
+, libGL ? null, libGLU ? null, wxGTK ? null, wxmac ? null, xorg ? null, wxSupport ? true
 , parallelBuild ? false
 , systemd
 }:
-
+let defaultWxSupport = wxSupport; in
```
Comment (jonringer) [inline pkgs/development/interpreters/erlang/generic-builder.nix:10]:
```suggestion
```

File: pkgs/development/interpreters/erlang/generic-builder.nix [hunk 2]
```diff
@@ -3,11 +3,11 @@
 # TODO: use jdk https://github.com/NixOS/nixpkgs/pull/89731
 , openjdk8 ? null # javacSupport
 , unixODBC ? null # odbcSupport
-, libGL ? null, libGLU ? null, wxGTK ? null, wxmac ? null, xorg ? null # wxSupport
+, libGL ? null, libGLU ? null, wxGTK ? null, wxmac ? null, xorg ? null, wxSupport ? true
```
Comment (jonringer) [inline pkgs/development/interpreters/erlang/generic-builder.nix:6]:
```suggestion
, libGL ? null, libGLU ? null, wxGTK ? null, wxmac ? null, xorg ? null
```

File: pkgs/development/interpreters/erlang/generic-builder.nix [hunk 3]
```diff
@@ -21,7 +21,7 @@
 , javacSupport ? false, javacPackages ? [ openjdk8 ]
 , odbcSupport ? false, odbcPackages ? [ unixODBC ]
 , withSystemd ? stdenv.isLinux # systemd support in epmd
-, wxSupport ? true, wxPackages ? [ libGL libGLU wxGTK xorg.libX11 ]
+, wxSupport ? defaultWxSupport, wxPackages ? [ libGL libGLU wxGTK xorg.libX11 ]
```
Comment (jonringer) [inline pkgs/development/interpreters/erlang/generic-builder.nix:24]:
```suggestion
, wxSupport ? true, wxPackages ? [ libGL libGLU wxGTK xorg.libX11 ]
```

**Review by jonringer — COMMENTED** [review]
My personal opinion is that it should be removed from the "bootstrapping" arguments

**Comment by yurrriq** [comment]
> My personal opinion is that it should be removed from the "bootstrapping" arguments

I agree.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 107394` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rabbitmq-server</li>
  </ul>
</details>

**Comment by happysalada** [comment]
@yorickvP What do you think about removing wxSupport from the bootstrapping argument?
I think this PR is almost there.

**Review by happysalada — APPROVED** [review]
🥇

**Comment by yorickvP** [comment]
Let me fix the build :)

**Comment by yorickvP** [comment]
So, it seems that the code doesn't work out nicely if it's not in the bootstrapping arguments (would have to override or change `callErlang`). Why even have bootstrapping arguments here in the first place?

**Comment by yorickvP** [comment]
The bootstrapping arguments are the second set of arguments in erlang's generic-builder.nix, confusing everyone. However, I think it's done now.

**Review by yurrriq — APPROVED** [review]

**Comment by happysalada** [comment]
@jonringer What do you think about the changes?

**Comment by happysalada** [comment]
@Mic92 sorry to ask you again. This PR's comment have been addressed and it has been reviewed by several people.
Feel free to update if you need anything.

**Comment by happysalada** [comment]
@Mic92 Thanks a lot!

**Pull request merged**